lib/Runtime/Debug/TTSnapObjects.cpp

```diff
@@ -2432,21 +2474,128 @@ namespace TTD
         }
 #endif
 
-        Js::RecyclableObject *DoObjectInflation_SnapAsyncFunction(const SnapObject *snpObject, InflateMap *inflator)
+        ////////////////////
+
+        Js::RecyclableObject* DoObjectInflation_SnapJavascriptPromiseAsyncSpawnExecutorFunction(const SnapObject *snpObject, InflateMap *inflator)
+        {
+            Js::ScriptContext *ctx = inflator->LookupScriptContext(snpObject->SnapType->ScriptContextLogId);
+            SnapJavascriptPromiseAsyncSpawnExecutorFunctionInfo* info = SnapObjectGetAddtlInfoAs<SnapJavascriptPromiseAsyncSpawnExecutorFunctionInfo *, SnapObjectType::JavascriptPromiseAsyncSpawnExecutorFunction>(snpObject);
+            Js::Var target = (info->target!= nullptr) ? inflator->InflateTTDVar(info->target) : nullptr;
+
+            Js::JavascriptGenerator* generator = nullptr;
+            if (info->generator != TTD_INVALID_PTR_ID)
+            {
+                generator = reinterpret_cast<Js::JavascriptGenerator*>(inflator->LookupObject(info->generator));
+            }
+
+            // TODO; why do we need to cast here??
```

Comment:
I agree; why is this cast required? The inheritance hierarchy looks like it should be fine. I assume there was some kind of compiler error that made you do this?

Reply:
Yeah, compiler was complaining.  I have no idea why...